.github/workflows/spi.yaml

```diff
@@ -0,0 +1,75 @@
+name: "spi build/test"
+on:
+  pull_request:
+    paths:
+      - "spi/**"
```

Comment:
probably wants to trigger on changes to `common/**` as well, both here and for the push trigger